packages/ts-sdk-apm/src/apm.ts

```diff
@@ -28,18 +15,22 @@ export default function createTelemetryMiddleware(
   function trace() {
     // validate apm and tracer
     if (!(options?.apm && typeof options.apm == 'function')) {
-      options.apm = defaultOptions.apm
+      console.warn(
+        'APM is missing or not a function. Proceeding without tracing.'
+      )
+      options.apm = () => {}
     }
 
     if (!(options?.tracer && typeof options.tracer == 'function')) {
-      options.tracer = defaultOptions.tracer
+      console.warn(
+        'Tracer is missing or not a function. Proceeding without tracing.'
+      )
+      options.tracer = () => {}
```

Comment:
I added a warning so it want totally stop the application

Reply:
Please can we remove any console statement from code, it is unnecessary. We cannot be logging statements in the customer's applications.

I believe on the review call we had last week on this, I explain to you what to do here. Just for clarification:

- Remove the console statement(s)
- Default these functions (tracer and apm) using an empty functions, so even if there is no attached collector the function can just fall through.

Reply:
so customers cannot see any traces, right?

Reply:
I mean you should default these to a default trace collector and remove the `console.warn` so even if the customer doesn't pass any tracer function as an option, we should still be able to record using our default tracer function, if you don't want to default it then fine, just go ahead and remove the console.warn.